models/protocol_tests/generate.go

```diff
@@ -391,6 +458,8 @@ func GenerateAssertions(out interface{}, shape *api.Shape, prefix string) string
 			return fmt.Sprintf("assert.Equal(t, %#v, string(%s))\n", out, prefix)
 		case "integer", "long":
 			return fmt.Sprintf("assert.Equal(t, int64(%#v), *%s)\n", out, prefix)
+		case "jsonvalue":
```

Comment:
Could you comment on checking for `case "jsonvalue"` both here and in the `case map[string]interface{}:` case above? When would both cases be needed?